ui/src/app/applications/components/application-resources-diff/application-resources-diff-section.tsx

```diff
@@ -0,0 +1,143 @@
+import * as React from 'react';
+import {addQueueItem, clearQueue} from './diff-queue';
+import {parseDiff} from 'react-diff-view';
+import {useVirtualizer} from '@tanstack/react-virtual';
+import {IndividualDiffSection} from './individual-diff-section';
+
+interface DiffFileModel {
+    loading: boolean;
+    file?: any;
+}
+
+export interface ApplicationResourcesDiffSectionProps {
+    prepareDiff: {
+        a: string;
+        b: string;
+        name: string;
+        hook?: boolean;
+    }[];
+    compactDiff: boolean;
+    inlineDiff: boolean;
+}
+
+export const ApplicationResourcesDiffSection = (props: ApplicationResourcesDiffSectionProps) => {
+    const {prepareDiff, compactDiff, inlineDiff} = props;
+    const parentRef = React.useRef();
+
+    const whiteBox = prepareDiff.length > 1 ? 'white-box' : '';
+    const showPath = prepareDiff.length > 1;
+    const viewType = inlineDiff ? 'unified' : 'split';
+
+    const [diffProgress, setDiffProgress] = React.useState({
+        finished: 0,
+        total: prepareDiff.length
+    });
+
+    const [diffFiles, setDiffFiles] = React.useState<DiffFileModel[]>(
+        prepareDiff.map(() => ({
+            loading: true,
+            file: {}
```

Comment:
Is it possible that hunks is undefined that is can pose a problem in the call to `markEdits` ?